packages/suite/src/views/wallet/coinmarket/savings/bank-account/index.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { Translation } from '@suite-components';
+import { Button, Input, Select } from '@trezor/components';
+import { InputError, KYCInProgress, withCoinmarket, WithCoinmarketProps } from '@wallet-components';
+import { getInputState } from '@wallet-utils/sendFormUtils';
+import { useCoinmarketSavingsBankAccount } from '@wallet-hooks/useCoinmarketSavingsBankAccount';
+import { Controller } from 'react-hook-form';
+import { UsBankAccountTypes } from '@wallet-constants/coinmarket/savings';
+import { buildUsBankAccountType } from '@wallet-utils/coinmarket/coinmarketUtils';
+
+const Header = styled.div`
+    font-size: 24px;
+    line-height: 24px;
+    margin-bottom: 31px;
+`;
+
+const BankAccountTypeNote = styled.div`
+    font-size: 14px;
+    color: ${props => props.theme.TYPE_LIGHT_GREY};
+    margin-bottom: 27px;
+`;
+
+const BankAccount = (props: WithCoinmarketProps) => {
+    const {
+        register,
+        errors,
+        getValues,
+        onSubmit,
+        handleSubmit,
+        formState,
+        isWatchingKYCStatus,
+        isSavingsTradeLoading,
+        control,
+    } = useCoinmarketSavingsBankAccount(props);
+
+    if (isSavingsTradeLoading) {
+        return <Translation id="TR_LOADING" />;
+    }
+    const typeSelectName = 'typeOption';
+    const nameInputName = 'name';
+    const routingNumberInputName = 'routingNumber';
+    const accountNumberInputName = 'accountNumber';
+    const bankAccountOwnerInputName = 'bankAccountOwner';
+    const { isSubmitting } = formState;
+    const { name, routingNumber, accountNumber, bankAccountOwner } = getValues();
+    const canSubmit = Object.keys(errors).length === 0 && !isSubmitting;
+    return (
+        <>
+            {isWatchingKYCStatus && <KYCInProgress />}
+            <Header>
+                <Translation id="TR_SAVINGS_BANK_ACCOUNT_HEADER" />
+            </Header>
+            <form onSubmit={handleSubmit(onSubmit)}>
+                <Controller
+                    control={control}
+                    name={typeSelectName}
+                    defaultValue={buildUsBankAccountType(UsBankAccountTypes[0])}
+                    render={({ onChange, value }) => (
```

Comment:
I would move the render function to the body of the component and used `useCallback` for clarity but not vital.